File: LeetCode/python3/442.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        dup = []
        for i in range(len(nums)):
            if nums[abs(nums[i]) - 1] < 0:
                dup.append(abs(nums[i]))
            else:
                nums[abs(nums[i]) - 1] *= -1
        return dup

File: LeetCode/cpp/442.cpp
class Solution {
public:
	vector<int> findDuplicates(vector<int>& nums) {
		int size = nums.size();
		vector<int> ret;
		for (int i = 0; i < size; ++i) {
			nums[abs(nums[i]) - 1] = -nums[abs(nums[i]) - 1];
			if (nums[abs(nums[i]) - 1] > 0)
				ret.push_back(abs(nums[i]));
		}
		return ret;
	}
};
